cloud/services/domains.go

```diff
@@ -62,6 +74,74 @@ func EnsureDNSEntries(ctx context.Context, mscope *scope.MachineScope, operation
 	return nil
 }
 
+// EnsureAkamaiDNSEntries ensures the domainrecord on Akamai EDGE DNS is created, updated, or deleted based on operation passed
+func EnsureAkamaiDNSEntries(ctx context.Context, mscope *scope.MachineScope, operation string, dnsEntries []DNSOptions) error {
+	// SetUp Client for Akamai EDGE DNS
+
+	fqdn := mscope.LinodeCluster.ObjectMeta.Name + "-" + mscope.LinodeCluster.Spec.Network.DNSUniqueIdentifier + "." + mscope.LinodeCluster.Spec.Network.DNSRootDomain
+
+	for _, dnsEntry := range dnsEntries {
+		recordBody, err := mscope.AkamaiDomainsClient.GetRecord(ctx, mscope.LinodeCluster.Spec.Network.DNSRootDomain, fqdn, string(dnsEntry.DNSRecordType))
+		if err != nil {
```

Comment:
do we know this will only error if the record doesn't exist?

Reply:
I've added an extra check now - This is what the API sends when the record is not found
```
2024/07/22 16:30:27 Title: Not Found; Type: https://problems.luna.akamaiapis.net/authoritative-dns/notFound; Detail: Unable to find record set with name nonexistent-domain.akafn.com and type A
```

Reply:
ah gotcha. If there's an error other than Not Found, should this bail out with the error so we can retry later (e.g. if the API is down?)

Reply:
Yeah you are right. We now bail out on any error other than "Not Found"